tests/merge-multiple-tests.js

```diff
@@ -0,0 +1,114 @@
+/* eslint-env mocha */
+const assert = require('assert');
+
+function multipleTests(merge) {
+  it('should override objects', function () {
+    const a = {
+      client: {
+        entry: './client.js'
+      }
+    };
+    const b = {
+      client: {
+        entry: './replaced.js'
+      }
+    };
+    const result = [
+      {
+        entry: './replaced.js'
+      }
+    ];
+
+    assert.deepEqual(merge(a, b), result);
+  });
+
+  it('should add new objects if not existing', function () {
+    const a = {
+      client: {
+        entry: './client.js'
+      },
+      server: {
+        entry: './server.js',
+      }
+    };
+    const b = {
+      client: {
+        entry: './replaced.js'
+      }
+    };
+    const result = [
+      {
+        entry: './replaced.js'
+      },
+      {
+        entry: './server.js'
+      }
+    ];
+
+    assert.deepEqual(merge(a, b), result);
+  });
+
+  it('should work with an array of objects', function () {
+    const a = {
+      client: {
+        entry: ['./client.js', './client2.js']
+      },
+      server: {
+        entry: ['./server.js', './server2.js']
+      }
+    };
+    const b = {
+      client: {
+        entry: ['./replaced.js', './replaced2.js']
+      }
+    };
+    const result = [
+      {
+        entry: ['./client.js', './client2.js', './replaced.js', './replaced2.js']
+      },
+      {
+        entry: ['./server.js', './server2.js']
+      }
+    ];
+
+    assert.deepEqual(merge(a, b), result);
+  });
+
+  it('should deeply merge objects', function () {
+    const a = {
+      client: {
+        entry: {
+          main: './client.js'
+        }
+      },
+      server: {
+        entry: {
+          main: './server.js'
+        }
+      }
+    };
+    const b = {
+      client: {
+        entry: {
+          main: './replaced.js'
+        }
+      }
+    };
+    const result = [
+      {
+        entry: {
+          main: './replaced.js'
+        }
+      },
+      {
+        entry: {
+          main: './server.js'
+        }
+      }
+    ];
+
+    assert.deepEqual(merge(a, b), result);
+  });
+}
+
```

Comment:
Extra test case to add. `a` with `client` and `b` with `server` alone. This should yield output with both configurations.

Test also partials like `a`: `client`, `server` + `b`: `server`, `test`.